webapp/src/app.tsx

```diff
@@ -1549,7 +1549,7 @@ ${compileService && compileService.githubCorePackage && compileService.gittag ?
     startTutorial(tutorialId: string, tutorialTitle?: string) {
         pxt.tickEvent("tutorial.start");
         // Check for Internet access
-        if (!Cloud.isOnline()) {
+        if (!pxt.Cloud.isNavigatorOnline()) {
```

Comment:
navigator.isOnline is not reliable. It does not handle hotel proxy, or disconnect internet. Why not using ``Cloud.isOnline``?

Reply:
Cloud online only works the second time you try to connect and it fails.

Reply:
This method still isn't fool proof, it's possible that you start the iframe, and not receive a response back since it never loaded. I'm trying to mitigate that a little, but there's no way for us to tell if an iframe didnt load (Unless we set a timeout to check, but then we're limiting the max it will take for a tutorial to load).

Reply:
> Cloud online only works the second time you try to connect and it fails.

I don't see why isNavigatorOnline would change that behavior. How do you repro?

Reply:
Try loading the page. Then disconnecting from the net. Cloud.isOnline maintains an _isOnline variable and so it will return true. 
IsNavigatorOnline asks the browser if it's online at that point in time, and would return false.

Reply:
We should fix this behavior rather than introducing a new fuction.

Reply:
isNavigatorOnline was already there. Cloud.isOnline serves a different purpose as it checks whether or not the backend is online. ie: both the user is connected to the internet AND they are able to access pxt.io/api and like I mentioned earlier, it's not calling the backend to check, it's just keeping a var around for whether or not it's online 
In the case of the tutorial, the backend is not the concern, but whether or not an iframe can connect to: 
/--docs#tutorial:tutorials/getting-started:js:en

I think the proper fix here is perhaps to send a message on load from the iframe telling the parent that we were able to connect. and same on the client side a timeout for say 5 seconds to check whether or not it received that initial onload message.

Reply:
Agreed on a better fix. Navigator.onLine does not mean the cloud is accessible, might be behind a hotel jail or blocked by the school.

/--docs is served by the cloud and uses the /api to request the .md file. This can fail in many places depending on when the connection drops.